types/packages/caver-contract/src/index.d.ts

```diff
@@ -0,0 +1,126 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import BN from 'bn.js'
+import { IWallet } from '../../caver-wallet/src'
+import { Result } from '../../caver-abi/src'
+import { KIP37 } from '../../caver-kct/src'
+import KIP7 from '../../caver-kct/src/kip7'
+import { TransactionReceipt, CallObject, BlockNumber, PromiEvent, LogsOptions } from '../../caver-core/src'
+import { Transaction, FeeDelegatedTransaction } from '../../caver-transaction/src'
+import { AbiItem } from '../../caver-utils/src'
+import KeyringContainer from '../../caver-wallet/src'
+
+export interface SendOptions {
+    from?: string
+    gasPrice?: number | string
+    gas?: number | string
+    data?: string
+    feeDelegation?: boolean
+    feePayer?: string
+    feeRatio?: number
+}
+
+export interface SendOptionsWithFormatter extends SendOptions {
+    contractDeployFormatter?: Function
+}
+
+export interface ContractOptions extends SendOptions {
+    address: string
+    jsonInterface: AbiItem[]
+}
+
+export interface ContractDeployParams {
+    data: string
+    arguments?: any[]
+}
+
+export interface EstimateGasOptions {
+    from?: string
+    gas?: number
+    value?: number | string | BN
+}
+
+export interface ContractMethod {
+    call(options?: CallObject, callback?: (err: Error, result: any) => void): Promise<any>
+    send(options: SendOptions, callback?: (err: Error, transactionHash: string) => void): PromiEvent<Contract>
+    sign(options: SendOptions): Promise<Transaction>
+    signAsFeePayer(options: SendOptions): Promise<Transaction>
+    estimateGas(options?: EstimateGasOptions, callback?: (err: Error, gas: number) => void): Promise<number>
+    encodeABI(): string
+}
+
+export interface EventOptions {
+    filter?: object
+    fromBlock?: number
+    toBlock?: number
+    topics?: string[]
+}
+
+export interface EventData {
+    returnValues?: {
+        [key: string]: any
+    }
+    raw: {
+        data?: string
+        topics?: string[]
+    }
+    event?: string
+    signature?: string
+    logIndex?: number
+    transactionIndex?: number
+    transactionHash?: string
+    blockHash?: string
+    blockNumber?: number
+    address?: string
+    id?: string
+}
+
+export default class Contract {
+    constructor(jsonInterface: AbiItem[], address?: string, options?: SendOptions)
+
+    _address: string
+    _jsonInterface: AbiItem[]
+    options: ContractOptions
+    defaultSendOptions: ContractOptions
+    defaultAccount: string | null
+    defaultBlock: BlockNumber
+    events: any
+    methods: any
+    _wallet: IWallet
+
+    clone(contractAddress?: string): Contract
+
+    setKeyrings(keyrings: KeyringContainer): void
```

Comment:
Do we need to this function?

Reply:
Yes, this is used to use Keyring instead of the Account(deprecated before common architecture feature) if keyrings are set.
This is used in inside logic, and new common architecture classes will call `setKeyring` when make an instance of the Contact to let instance know that have to use Keyring instead of Account.